pkg/ebpf/net_events.go

```diff
@@ -186,14 +189,18 @@ func netPacketProtocolHandler(buffer []byte, evtMeta EventMeta, ctx procinfo.Pro
 	return evt, nil
 }
 
-func parseCaptureData(payload []byte) (CaptureData, int, error) {
+func parseCaptureData(payload []byte, debug bool) (CaptureData, error) {
 	var capData CaptureData
 	if len(payload) < 8 {
-		return capData, 0, fmt.Errorf("payload too short\n")
+		return capData, fmt.Errorf("payload too short\n")
 	}
-	capData.PacketLen = binary.LittleEndian.Uint32(payload[0:4])
+	capData.PacketLength = binary.LittleEndian.Uint32(payload[0:4])
 	capData.InterfaceIndex = binary.LittleEndian.Uint32(payload[4:8])
-	return capData, 8, nil
+	capData.PacketStartOffset = 8
+	if debug {
+		capData.PacketStartOffset = 48
```

Comment:
Out of curiosity, what is it about debug mode that pushes the packet start offset ahead? What is it exactly that's 'in debug mode'?

Reply:
This shouldn't happen here, see my previous comments.
In addition, we shuoldn't use magic numbers, as these might change in the future if and when we will change the data passed from bpf code.

Answer to @grantseltzer question: The packet payload comes after the metadata.
If debug is not enabled, only the packet length and interface index are passed.
Otherwise, there is other metadata (source ip, port, etc) that is passed

Reply:
just to elaborate: 

in debug mode we send extra data for debug.

for each network event, the first bytes are:
"Timestamp:NetworkEventId:HostTid:ProcessName".

if the NetworkEventId indicates a packet, we add:
"PacketLen:InterfaceIndex"

and for debug mode we add:
"SrcIp:DstIP:SrcPort:DstPort:Protocol:fullPacket"

so packet with no debug:
"Timestamp:NetworkEventId:HostTid:ProcessName:PacketLen:InterfaceIndex:fullPacket"

packet with debug mode:
"Timestamp:NetworkEventId:HostTid:ProcessName:PacketLen:InterfaceIndex:SrcIp:DstIP:SrcPort:DstPort:Protocol:fullPacket"